api/users/permissions.py

```diff
@@ -47,3 +49,63 @@ def has_permission(self, request, view):
     def has_object_permission(self, request, view, obj):
         assert isinstance(obj, OSFUser), f'obj must be a User, got {obj}'
         return not obj.is_registered
+
+
+class UserMessagePermissions(permissions.BasePermission):
+    """
+    Custom permission to allow only institutional admins to create certain types of UserMessages.
+    """
+    def has_permission(self, request, view) -> bool:
+        """
+        Validate if the user has permission to perform the requested action.
+        Args:
+            request: The HTTP request.
+            view: The view handling the request.
+        Returns:
+            bool: True if the user has the required permission, False otherwise.
+        """
+        if request.method != 'POST':
+            return False
+
+        user = request.user
+        if not user or user.is_anonymous:
+            return False
+
+        message_type = request.data.get('message_type')
+        if message_type == MessageTypes.INSTITUTIONAL_REQUEST:
+            return self._validate_institutional_request(request, user)
```

Comment:
Is this the right place for this check? Again, it would give a 403, when a 409 Conflict or just a straight 400 might be more appropriate.

Reply:
Since currently this only applies to insti admin, I'll make that check here and move the institution validation and other details to the serializer. Such as:
```
        if message_type == MessageTypes.INSTITUTIONAL_REQUEST:
            return user.is_institutional_admin
        else:
            raise NotImplementedError('only request type')
```
Good comment thanks.